From a2f97764f9695cf6f10a1c9f21f0e89a59a46bb0 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Fri, 11 Dec 2020 15:52:41 +0700
Subject: [PATCH 350/357] arm64: dts: renesas: r8a774c0: reconstruct G2E device
 files

This commit reconstructs G2E device files support:
- Update differences in each revisions.
- Create new device files to support G2E Rev.C with suffix "revc".
- Use current "r8a774c0-[cat874,ek874].dtb " to support G2E Rev.[D,E]
and later.
- Update sdhi channel of wifi lines for each revisions.
- Update bluetooth enable pin for each revisions.
- Update backlight enable pin for Rev.[D,E].
- Remove Switch selecting between LVDS and MIPI-DSI controlled by GP0_17
in Rev.[D,E].
- Update LED2 GPIO pin for each revisions.
- Add audio support for Rev.[A,B,C].

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 arch/arm64/boot/dts/renesas/Makefile               |   7 +-
 .../boot/dts/renesas/r8a774c0-cat874-revc.dts      | 260 +++++++++
 arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts    | 583 +--------------------
 .../boot/dts/renesas/r8a774c0-ek874-idk-2121wr.dts |  19 +-
 .../boot/dts/renesas/r8a774c0-ek874-mipi-2.1.dts   |   2 +-
 .../dts/renesas/r8a774c0-ek874-revc-idk-2121wr.dts | 116 ++++
 .../dts/renesas/r8a774c0-ek874-revc-mipi-2.1.dts   |  58 ++
 .../arm64/boot/dts/renesas/r8a774c0-ek874-revc.dts |  15 +
 arch/arm64/boot/dts/renesas/r8a774c0-ek874.dts     |   1 +
 .../boot/dts/renesas/r8a774c0-es10-cat874.dts      | 441 ++--------------
 10 files changed, 523 insertions(+), 979 deletions(-)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774c0-cat874-revc.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-idk-2121wr.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-mipi-2.1.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc.dts

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index e8ed4e3..089c303 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -19,9 +19,14 @@ dtb-$(CONFIG_ARCH_R8A774B1) += r8a774b1-hihope-rzg2n-ex-idk-1110wr.dtb \
 			       r8a774b1-hihope-rzg2n-rev2-ex-idk-1110wr.dtb
 dtb-$(CONFIG_ARCH_R8A774B1) += r8a774b1-hihope-rzg2n-ex-mipi-2.1.dtb \
 			       r8a774b1-hihope-rzg2n-rev2-ex-mipi-2.1.dtb
-dtb-$(CONFIG_ARCH_R8A774C0) += r8a774c0-cat874.dtb r8a774c0-ek874.dtb \
+dtb-$(CONFIG_ARCH_R8A774C0) += r8a774c0-cat874.dtb \
+			       r8a774c0-ek874.dtb \
 			       r8a774c0-ek874-idk-2121wr.dtb \
 			       r8a774c0-ek874-mipi-2.1.dtb
+dtb-$(CONFIG_ARCH_R8A774C0) += r8a774c0-cat874-revc.dtb \
+			       r8a774c0-ek874-revc.dtb \
+			       r8a774c0-ek874-revc-idk-2121wr.dtb \
+			       r8a774c0-ek874-revc-mipi-2.1.dtb
 dtb-$(CONFIG_ARCH_R8A774C0) += r8a774c0-es10-cat874.dtb \
 			       r8a774c0-es10-ek874.dtb \
 			       r8a774c0-es10-ek874-idk-2121wr.dtb \
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-cat874-revc.dts b/arch/arm64/boot/dts/renesas/r8a774c0-cat874-revc.dts
new file mode 100644
index 0000000..00a3ebc
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-cat874-revc.dts
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the Silicon Linux RZ/G2E 96board platform (CAT874)
+ * Rev.C
+ *
+ * Copyright (C) 2019 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a774c0.dtsi"
+#define WIFI_PARENT_SDHI	sdhi3
+#include "cat874-common.dtsi"
+#include <dt-bindings/display/tda998x.h>
+
+/ {
+	hdmi-out {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi_con_out: endpoint {
+				remote-endpoint = <&tda19988_out>;
+			};
+		};
+	};
+
+	leds {
+		led2 {
+			gpios = <&gpio4 10 GPIO_ACTIVE_HIGH>;
+			label = "LED2";
+		};
+	};
+
+	sound: sound {
+		compatible = "simple-audio-card";
+
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "rcar_sound";
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		simple-audio-card,dai-link@0 {
+			reg = <0>;
+			format = "i2s";
+			cpu {
+				sound-dai = <&rcar_sound 0>;
+				bitclock-master;
+				frame-master;
+			};
+
+			codec {
+				sound-dai = <&tda19988>;
+			};
+		};
+
+		simple-audio-card,dai-link@1 {
+			reg = <1>;
+			format = "i2s";
+			cpu {
+				sound-dai = <&rcar_sound 1>;
+				bitclock-master;
+				frame-master;
+			};
+
+			codec {
+				sound-dai = <&pcm5102a>;
+			};
+		};
+	};
+
+	pcm5102a: pcm5102a {
+		compatible = "ti,pcm5102a";
+		#sound-dai-cells = <0>;
+	};
+};
+
+&audio_clk_a {
+	clock-frequency = <22579200>;
+};
+
+&sound {
+	simple-audio-card,dai-link@1 {
+		reg = <1>;
+		format = "i2s";
+		cpu {
+			sound-dai = <&rcar_sound 1>;
+			bitclock-master;
+			frame-master;
+		};
+
+		codec {
+			sound-dai = <&tda19988>;
+		};
+	};
+};
+
+&du {
+	ports {
+		port@0 {
+			endpoint {
+				remote-endpoint = <&tda19988_in>;
+			};
+		};
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+
+	hd3ss3220@47 {
+		compatible = "ti,hd3ss3220";
+		reg = <0x47>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <3 IRQ_TYPE_LEVEL_LOW>;
+
+		connector {
+			compatible = "usb-c-connector";
+			label = "USB-C";
+			data-role = "dual";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@1 {
+					reg = <1>;
+					hd3ss3220_ep: endpoint {
+						remote-endpoint = <&usb3_role_switch>;
+					};
+				};
+			};
+		};
+	};
+
+	tda19988: tda19988@70 {
+		compatible = "nxp,tda998x";
+		reg = <0x70>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <1 IRQ_TYPE_LEVEL_LOW>;
+
+		video-ports = <0x234501>;
+
+		#sound-dai-cells = <0>;
+		audio-ports = <TDA998x_I2S 0x03>;
+		clocks = <&rcar_sound 1>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				tda19988_in: endpoint {
+					remote-endpoint = <&du_out_rgb>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				tda19988_out: endpoint {
+					remote-endpoint = <&hdmi_con_out>;
+				};
+			};
+		};
+	};
+};
+
+&rcar_sound {
+	pinctrl-0 = <&sound_pins &sound_clk_pins>;
+	pinctrl-names = "default";
+
+	/* Single DAI */
+	#sound-dai-cells = <1>;
+
+	/* audio_clkout0/1/2/3 */
+	#clock-cells = <1>;
+	clock-frequency = <11289600>;
+
+	status = "okay";
+
+	rcar_sound,dai {
+		dai0 {
+			playback = <&ssi0 &src0 &dvc0>;
+		};
+
+		dai1 {
+			playback = <&ssi4 &src4 &dvc1>;
+		};
+	};
+
+};
+
+&ssi4 {
+	shared-pin;
+};
+
+&ohci0 {
+	dr_mode = "host";
+};
+
+&ehci0 {
+	dr_mode = "host";
+};
+
+&usb2_phy0 {
+	renesas,no-otg-pins;
+};
+
+&usb3_peri0 {
+	usb-role-switch;
+
+	port {
+		usb3_role_switch: endpoint {
+			remote-endpoint = <&hd3ss3220_ep>;
+		};
+	};
+};
+
+&xhci0 {
+	pinctrl-0 = <&usb30_pins>;
+	pinctrl-names = "default";
+};
+
+&sdhi3 {
+	pinctrl-0 = <&sdhi3_pins>;
+	pinctrl-names = "default";
+};
+
+&hscif2 {
+	bluetooth {
+		enable-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&pfc {
+	sdhi3_pins: sd3 {
+		groups = "sdhi3_data4", "sdhi3_ctrl";
+		function = "sdhi3";
+		power-source = <1800>;
+	};
+
+	usb30_pins: usb30 {
+		groups = "usb30", "usb30_id";
+		function = "usb30";
+	};
+
+	sound_clk_pins: sound_clk {
+		groups = "audio_clkout1_a",
+			 "audio_clk_a", "audio_clk_b_a", "audio_clk_c_a";
+		function = "audio_clk";
+	};
+
+	sound_pins: sound {
+		groups = "ssi01239_ctrl", "ssi0_data",
+			 "ssi349_ctrl", "ssi3_data", "ssi4_data";
+		function = "ssi";
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts b/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
index df5730f..220a5b4 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-cat874.dts
@@ -1,599 +1,40 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
  * Device Tree Source for the Silicon Linux RZ/G2E 96board platform (CAT874)
+ * Rev.[D,E]
  *
  * Copyright (C) 2019 Renesas Electronics Corp.
  */
 
 /dts-v1/;
 #include "r8a774c0.dtsi"
-#include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/display/tda998x.h>
+#define WIFI_PARENT_SDHI	sdhi1
+#include "cat874-common.dtsi"
 
 / {
-	model = "Silicon Linux RZ/G2E 96board platform (CAT874)";
-	compatible = "si-linux,cat874", "renesas,r8a774c0";
-
-	aliases {
-		serial0 = &scif2;
-		serial1 = &hscif2;
-		serial2 = &hscif3;
-		serial3 = &hscif4;
-		spi0 = &msiof0;
-	};
-
-	chosen {
-		bootargs = "ignore_loglevel rw root=/dev/nfs ip=on";
-		stdout-path = "serial0:115200n8";
-	};
-
-	hdmi-out {
-		compatible = "hdmi-connector";
-		type = "a";
-
-		port {
-			hdmi_con_out: endpoint {
-				remote-endpoint = <&tda19988_out>;
-			};
-		};
-	};
-
 	leds {
-		compatible = "gpio-leds";
-
-		led0 {
-			gpios = <&gpio5 19 GPIO_ACTIVE_HIGH>;
-			label = "LED0";
-		};
-
-		led1 {
-			gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>;
-			label = "LED1";
-		};
-
 		led2 {
-			gpios = <&gpio4 10 GPIO_ACTIVE_HIGH>;
+			gpios = <&gpio5 16 GPIO_ACTIVE_HIGH>;
 			label = "LED2";
 		};
-
-		led3 {
-			gpios = <&gpio6 4 GPIO_ACTIVE_HIGH>;
-			label = "LED3";
-		};
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x78000000>;
-	};
-
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		mmp_reserved: linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x68000000 0x0 0x08000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
-	reg_12p0v: regulator-12p0v {
-		compatible = "regulator-fixed";
-		regulator-name = "D12.0V";
-		regulator-min-microvolt = <12000000>;
-		regulator-max-microvolt = <12000000>;
-		regulator-boot-on;
-		regulator-always-on;
-	};
-
-	sound: sound {
-		compatible = "simple-audio-card";
-
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rcar_sound";
-
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		simple-audio-card,dai-link@0 {
-			reg = <0>;
-			format = "i2s";
-			cpu {
-				sound-dai = <&rcar_sound 0>;
-				bitclock-master;
-				frame-master;
-			};
-
-			codec {
-				sound-dai = <&tda19988>;
-			};
-		};
-
-		simple-audio-card,dai-link@1 {
-			reg = <1>;
-			format = "i2s";
-			cpu {
-				sound-dai = <&rcar_sound 1>;
-				bitclock-master;
-				frame-master;
-			};
-
-			codec {
-				sound-dai = <&pcm5102a>;
-			};
-		};
-	};
-
-	pcm5102a: pcm5102a {
-		compatible = "ti,pcm5102a";
-		#sound-dai-cells = <0>;
-	};
-
-	vcc_sdhi0: regulator-vcc-sdhi0 {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI0 Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-		regulator-boot-on;
-	};
-
-	vccq_sdhi0: regulator-vccq-sdhi0 {
-		compatible = "regulator-gpio";
-
-		regulator-name = "SDHI0 VccQ";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpios = <&gpio3 13 GPIO_ACTIVE_HIGH>;
-		gpios-states = <1>;
-		states = <3300000 1
-			  1800000 0>;
-	};
-
-	wlan_en_reg: fixedregulator {
-		compatible = "regulator-fixed";
-		regulator-name = "wlan-en-regulator";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		startup-delay-us = <70000>;
-
-		gpio = <&gpio2 25 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
 	};
-
-	x13_clk: x13 {
-		compatible = "fixed-clock";
-		#clock-cells = <0>;
-		clock-frequency = <74250000>;
-	};
-};
-
-&audio_clk_a {
-	clock-frequency = <22579200>;
 };
 
-&du {
-	pinctrl-0 = <&du_pins>;
+&sdhi1 {
+	pinctrl-0 = <&sdhi1_pins>;
 	pinctrl-names = "default";
-	status = "okay";
-
-	clocks = <&cpg CPG_MOD 724>,
-		 <&cpg CPG_MOD 723>,
-		 <&x13_clk>;
-	clock-names = "du.0", "du.1", "dclkin.0";
-
-	ports {
-		port@0 {
-			endpoint {
-				remote-endpoint = <&tda19988_in>;
-			};
-		};
-	};
-};
-
-&ehci0 {
-	dr_mode = "host";
-	status = "okay";
-
-	memory-region = <&global_cma>;
-};
-
-&extal_clk {
-	clock-frequency = <48000000>;
-};
-
-&hscif2 {
-	pinctrl-0 = <&hscif2_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-
-	bluetooth {
-		compatible = "ti,wl1837-st";
-		enable-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
-	};
-};
-
-&hscif3 {
-	pinctrl-0 = <&hscif3_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-};
-
-&hscif4 {
-	pinctrl-0 = <&hscif4_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-};
-
-&i2c0 {
-	status = "okay";
-	clock-frequency = <100000>;
-
-	hd3ss3220@47 {
-		compatible = "ti,hd3ss3220";
-		reg = <0x47>;
-		interrupt-parent = <&gpio6>;
-		interrupts = <3 IRQ_TYPE_LEVEL_LOW>;
-
-		connector {
-			compatible = "usb-c-connector";
-			label = "USB-C";
-			data-role = "dual";
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@1 {
-					reg = <1>;
-					hd3ss3220_ep: endpoint {
-						remote-endpoint = <&usb3_role_switch>;
-					};
-				};
-			};
-		};
-	};
-
-	tda19988: tda19988@70 {
-		compatible = "nxp,tda998x";
-		reg = <0x70>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <1 IRQ_TYPE_LEVEL_LOW>;
-
-		video-ports = <0x234501>;
-
-		#sound-dai-cells = <0>;
-		audio-ports = <TDA998x_I2S 0x03>;
-		clocks = <&rcar_sound 1>;
-
-		ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				tda19988_in: endpoint {
-					remote-endpoint = <&du_out_rgb>;
-				};
-			};
-
-			port@1 {
-				reg = <1>;
-				tda19988_out: endpoint {
-					remote-endpoint = <&hdmi_con_out>;
-				};
-			};
-		};
-	};
-};
-
-&i2c1 {
-	pinctrl-0 = <&i2c1_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	rtc@32 {
-		compatible = "epson,rx8571";
-		reg = <0x32>;
-	};
-};
-
-&lvds0 {
-	status = "okay";
-
-	clocks = <&cpg CPG_MOD 727>, <&x13_clk>, <&extal_clk>;
-	clock-names = "fck", "dclkin.0", "extal";
-};
-
-&ohci0 {
-	dr_mode = "host";
-	status = "okay";
-
-	memory-region = <&global_cma>;
-};
-
-&pcie_bus_clk {
-	clock-frequency = <100000000>;
-};
-
-&pciec0 {
-	/* Map all possible DDR as inbound ranges */
-	dma-ranges = <0x42000000 0 0x40000000 0 0x40000000 0 0x80000000>;
 };
 
 &pfc {
-	du_pins: du {
-		groups = "du_rgb888", "du_clk_out_0", "du_sync", "du_disp",
-			 "du_clk_in_0";
-		function = "du";
-	};
-
-	hscif2_pins: hscif2 {
-		groups = "hscif2_data_a", "hscif2_ctrl_a";
-		function = "hscif2";
-	};
-
-	hscif3_pins: hscif3 {
-		groups = "hscif3_data_c", "hscif3_ctrl_c";
-		function = "hscif3";
-	};
-
-	hscif4_pins: hscif4 {
-		groups = "hscif4_data_b";
-		function = "hscif4";
-	};
-
-	i2c1_pins: i2c1 {
-		groups = "i2c1_b";
-		function = "i2c1";
-	};
-
-	scif2_pins: scif2 {
-		groups = "scif2_data_a";
-		function = "scif2";
-	};
-
-	sdhi0_pins: sd0 {
-		groups = "sdhi0_data4", "sdhi0_ctrl";
-		function = "sdhi0";
-		power-source = <3300>;
-	};
-
-	sdhi0_pins_uhs: sd0_uhs {
-		groups = "sdhi0_data4", "sdhi0_ctrl";
-		function = "sdhi0";
-		power-source = <1800>;
-	};
-
-	sdhi3_pins: sd3 {
-		groups = "sdhi3_data4", "sdhi3_ctrl";
-		function = "sdhi3";
+	sdhi1_pins: sd1 {
+		groups = "sdhi1_data4", "sdhi1_ctrl";
+		function = "sdhi1";
 		power-source = <1800>;
 	};
-
-	sound_clk_pins: sound_clk {
-		groups = "audio_clkout1_a",
-			 "audio_clk_a", "audio_clk_b_a", "audio_clk_c_a";
-		function = "audio_clk";
-	};
-
-	sound_pins: sound {
-		groups = "ssi01239_ctrl", "ssi0_data",
-			 "ssi349_ctrl", "ssi3_data", "ssi4_data";
-		function = "ssi";
-	};
-
-	usb30_pins: usb30 {
-		groups = "usb30", "usb30_id";
-		function = "usb30";
-	};
-
-	pwm6_pins: pwm6 {
-		groups = "pwm6_a";
-		function = "pwm6";
-	};
-
-	tpu0_pins: tpu {
-		groups = "tpu0_to0", "tpu0_to1";
-		function = "tpu0";
-	};
-
-	msiof0_pins: spi0 {
-		groups = "msiof0_clk", "msiof0_sync",
-			 "msiof0_rxd", "msiof0_txd";
-		function = "msiof0";
-	};
-};
-
-&rcar_sound {
-	pinctrl-0 = <&sound_pins &sound_clk_pins>;
-	pinctrl-names = "default";
-
-	/* Single DAI */
-	#sound-dai-cells = <1>;
-
-	/* audio_clkout0/1/2/3 */
-	#clock-cells = <1>;
-	clock-frequency = <11289600>;
-
-	status = "okay";
-
-	rcar_sound,dai {
-		dai0 {
-			playback = <&ssi0 &src0 &dvc0>;
-		};
-
-		dai1 {
-			playback = <&ssi4 &src4 &dvc1>;
-		};
-	};
-};
-
-&ssi4 {
-	shared-pin;
-};
-
-&rwdt {
-	timeout-sec = <60>;
-	status = "okay";
-};
-
-&scif2 {
-	pinctrl-0 = <&scif2_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-};
-
-&sdhi0 {
-	pinctrl-0 = <&sdhi0_pins>;
-	pinctrl-1 = <&sdhi0_pins_uhs>;
-	pinctrl-names = "default", "state_uhs";
-
-	vmmc-supply = <&vcc_sdhi0>;
-	vqmmc-supply = <&vccq_sdhi0>;
-	cd-gpios = <&gpio3 12 GPIO_ACTIVE_LOW>;
-	bus-width = <4>;
-	sd-uhs-sdr50;
-	sd-uhs-sdr104;
-	status = "okay";
-};
-
-&sdhi3 {
-	status = "okay";
-	pinctrl-0 = <&sdhi3_pins>;
-	pinctrl-names = "default";
-
-	vmmc-supply = <&wlan_en_reg>;
-	bus-width = <4>;
-	non-removable;
-	cap-power-off-card;
-	keep-power-in-suspend;
-
-	#address-cells = <1>;
-	#size-cells = <0>;
-	wlcore: wlcore@2 {
-		compatible = "ti,wl1837";
-		reg = <2>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>;
-	};
-};
-
-&usb2_phy0 {
-	renesas,no-otg-pins;
-	status = "okay";
 };
 
-&usb3_peri0 {
-	companion = <&xhci0>;
-	status = "okay";
-	usb-role-switch;
-
-	port {
-		usb3_role_switch: endpoint {
-			remote-endpoint = <&hd3ss3220_ep>;
-		};
+&hscif2 {
+	bluetooth {
+		enable-gpios = <&gpio0 17 GPIO_ACTIVE_HIGH>;
 	};
 };
-
-&xhci0 {
-	pinctrl-0 = <&usb30_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-
-	memory-region = <&global_cma>;
-};
-
-&cmt0 {
-	status = "okay";
-};
-
-&cmt1 {
-	status = "okay";
-};
-
-&tmu0 {
-	status = "okay";
-};
-
-&tmu1 {
-	status = "okay";
-};
-
-&tmu2 {
-	status = "okay";
-};
-
-&tmu3 {
-	status = "okay";
-};
-
-&tmu4 {
-	status = "okay";
-};
-
-&pwm6 {
-	pinctrl-0 = <&pwm6_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-};
-
-&vspb {
-	status = "okay";
-};
-
-&vspi0 {
-	status = "okay";
-};
-
-&tpu{
-	pinctrl-0 = <&tpu0_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-};
-
-&msiof0 {
-	pinctrl-0 = <&msiof0_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-};
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-idk-2121wr.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-idk-2121wr.dts
index a7b27d0..88ac9db 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-idk-2121wr.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-idk-2121wr.dts
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Device Tree Source for the Silicon Linux RZ/G2E evaluation kit (EK874),
- * connected to an Advantech IDK-2121WR 21.5" LVDS panel
+ * Device Tree Source for the Silicon Linux RZ/G2E evaluation kit (EK874)
+ * Rev.[D,E], connected to an Advantech IDK-2121WR 21.5" LVDS panel
  *
  * Copyright (C) 2019 Renesas Electronics Corp.
  */
@@ -17,7 +17,7 @@
 		default-brightness-level = <6>;
 
 		power-supply = <&reg_12p0v>;
-		enable-gpios = <&gpio6 12 GPIO_ACTIVE_HIGH>;
+		enable-gpios = <&gpio3 15 GPIO_ACTIVE_HIGH>;
 	};
 
 	panel-lvds {
@@ -63,19 +63,6 @@
 	};
 };
 
-&gpio0 {
-	/*
-	 * When GP0_17 is low LVDS[01] are connected to the LVDS connector
-	 * When GP0_17 is high LVDS[01] are connected to the LT8918L
-	 */
-	lvds-connector-en-gpio{
-		gpio-hog;
-		gpios = <17 GPIO_ACTIVE_HIGH>;
-		output-low;
-		line-name = "lvds-connector-en-gpio";
-	};
-};
-
 &lvds0 {
 	ports {
 		port@1 {
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-mipi-2.1.dts
index b6b0ae1..12c52a9 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-mipi-2.1.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-mipi-2.1.dts
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
  * Device Tree Source for the Silicon Linux RZ/G2E 96board platform (CAT874)
- * connected with aistarvision-mipi-v2-adapter board
+ * Rev.[D,E] connected with aistarvision-mipi-v2-adapter board
  *
  * Copyright (C) 2020 Renesas Electronics Corp.
  */
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-idk-2121wr.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-idk-2121wr.dts
new file mode 100644
index 0000000..a2c1a8b
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-idk-2121wr.dts
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the Silicon Linux RZ/G2E evaluation kit (EK874)
+ * Rev.C, connected to an Advantech IDK-2121WR 21.5" LVDS panel
+ *
+ * Copyright (C) 2019 Renesas Electronics Corp.
+ */
+
+#include "r8a774c0-ek874-revc.dts"
+
+/ {
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm5 0 50000>;
+
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+
+		power-supply = <&reg_12p0v>;
+		enable-gpios = <&gpio6 12 GPIO_ACTIVE_HIGH>;
+	};
+
+	panel-lvds {
+		compatible = "advantech,idk-2121wr", "panel-lvds";
+
+		width-mm = <476>;
+		height-mm = <268>;
+
+		data-mapping = "vesa-24";
+
+		panel-timing {
+			clock-frequency = <148500000>;
+			hactive = <1920>;
+			vactive = <1080>;
+			hsync-len = <44>;
+			hfront-porch = <88>;
+			hback-porch = <148>;
+			vfront-porch = <4>;
+			vback-porch = <36>;
+			vsync-len = <5>;
+		};
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				dual-lvds-odd-pixels;
+				panel_in0: endpoint {
+					remote-endpoint = <&lvds0_out>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				dual-lvds-even-pixels;
+				panel_in1: endpoint {
+					remote-endpoint = <&lvds1_out>;
+				};
+			};
+		};
+	};
+};
+
+&gpio0 {
+	/*
+	 * When GP0_17 is low LVDS[01] are connected to the LVDS connector
+	 * When GP0_17 is high LVDS[01] are connected to the LT8918L
+	 */
+	lvds-connector-en-gpio{
+		gpio-hog;
+		gpios = <17 GPIO_ACTIVE_HIGH>;
+		output-low;
+		line-name = "lvds-connector-en-gpio";
+	};
+};
+
+&lvds0 {
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
+				remote-endpoint = <&panel_in0>;
+			};
+		};
+	};
+};
+
+&lvds1 {
+	status = "okay";
+
+	clocks = <&cpg CPG_MOD 727>, <&x13_clk>, <&extal_clk>;
+	clock-names = "fck", "dclkin.0", "extal";
+
+	ports {
+		port@1 {
+			lvds1_out: endpoint {
+				remote-endpoint = <&panel_in1>;
+			};
+		};
+	};
+};
+
+&pfc {
+	pwm5_pins: pwm5 {
+		groups = "pwm5_a";
+		function = "pwm5";
+	};
+};
+
+&pwm5 {
+	pinctrl-0 = <&pwm5_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-mipi-2.1.dts
new file mode 100644
index 0000000..a01e378
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc-mipi-2.1.dts
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the Silicon Linux RZ/G2E 96board platform (CAT874)
+ * Rev.C connected with aistarvision-mipi-v2-adapter board
+ *
+ * Copyright (C) 2020 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r8a774c0-ek874-revc.dts"
+#define MIPI_PARENT_I2C_CSI40 i2c3
+#include "aistarvision-mipi-adapter-2.1.dtsi"
+
+/ {
+	model = "Silicon Linux RZ/G2E evaluation kit EK874 (CAT874 + CAT875) with aistarvision-mipi-v2-adapter board";
+	compatible = "si-linux,cat875", "si-linux,cat874", "renesas,r8a774c0";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&vin4 {
+	status = "okay";
+};
+
+&vin5 {
+	status = "okay";
+};
+
+&csi40 {
+	status = "okay";
+
+	ports {
+		port {
+			csi40_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi40_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi40 {
+	enable-gpios = <&gpio5 5 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio5 3 GPIO_ACTIVE_LOW>;
+
+	virtual-channel = /bits/ 8 <0>;
+
+	port {
+		ov5645_csi40_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi40_in>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc.dts
new file mode 100644
index 0000000..741c3c5
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874-revc.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the Silicon Linux RZ/G2E evaluation kit (EK874)
+ * Rev.C
+ *
+ * Copyright (C) 2020 Renesas Electronics Corp.
+ */
+
+#include "r8a774c0-cat874-revc.dts"
+#include "cat875.dtsi"
+
+/ {
+	model = "Silicon Linux RZ/G2E evaluation kit EK874 (CAT874 + CAT875)";
+	compatible = "si-linux,cat875", "si-linux,cat874", "renesas,r8a774c0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-ek874.dts b/arch/arm64/boot/dts/renesas/r8a774c0-ek874.dts
index e7b6619..822e4a1 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-ek874.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-ek874.dts
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
  * Device Tree Source for the Silicon Linux RZ/G2E evaluation kit (EK874)
+ * Rev.[D,E]
  *
  * Copyright (C) 2019 Renesas Electronics Corp.
  */
diff --git a/arch/arm64/boot/dts/renesas/r8a774c0-es10-cat874.dts b/arch/arm64/boot/dts/renesas/r8a774c0-es10-cat874.dts
index ba1d3a9..2bdd7d3 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0-es10-cat874.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774c0-es10-cat874.dts
@@ -1,32 +1,18 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Device Tree Source for the Silicon Linux RZ/G2E 96board platform (CAT874)
+ * Device Tree Source for the Silicon Linux RZ/G2E ES1.0 96board platform
+ * (CAT874)
  *
- * Copyright (C) 2019 Renesas Electronics Corp.
+ * Copyright (C) 2020 Renesas Electronics Corp.
  */
 
 /dts-v1/;
 #include "r8a774c0-es10.dtsi"
-#include <dt-bindings/gpio/gpio.h>
+#define WIFI_PARENT_SDHI	sdhi3
+#include "cat874-common.dtsi"
 #include <dt-bindings/display/tda998x.h>
 
 / {
-	model = "Silicon Linux RZ/G2E 96board platform (CAT874)";
-	compatible = "si-linux,cat874", "renesas,r8a774c0";
-
-	aliases {
-		serial0 = &scif2;
-		serial1 = &hscif2;
-		serial2 = &hscif3;
-		serial3 = &hscif4;
-		spi0 = &msiof0;
-	};
-
-	chosen {
-		bootargs = "ignore_loglevel rw root=/dev/nfs ip=on";
-		stdout-path = "serial0:115200n8";
-	};
-
 	hdmi-out {
 		compatible = "hdmi-connector";
 		type = "a";
@@ -39,76 +25,10 @@
 	};
 
 	leds {
-		compatible = "gpio-leds";
-
-		led0 {
-			gpios = <&gpio5 19 GPIO_ACTIVE_HIGH>;
-			label = "LED0";
-		};
-
-		led1 {
-			gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>;
-			label = "LED1";
-		};
-
 		led2 {
 			gpios = <&gpio4 10 GPIO_ACTIVE_HIGH>;
 			label = "LED2";
 		};
-
-		led3 {
-			gpios = <&gpio6 4 GPIO_ACTIVE_HIGH>;
-			label = "LED3";
-		};
-	};
-
-	memory@48000000 {
-		device_type = "memory";
-		/* first 128MB is reserved for secure area. */
-		reg = <0x0 0x48000000 0x0 0x78000000>;
-	};
-
-	reserved-memory {
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
-
-		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x10000000>;
-			linux,cma-default;
-		};
-
-		/* device specific region for contiguous allocations */
-		mmp_reserved: linux,multimedia {
-			compatible = "shared-dma-pool";
-			reusable;
-			reg = <0x00000000 0x68000000 0x0 0x08000000>;
-		};
-	};
-
-	mmngr {
-		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>;
-	};
-
-	mmngrbuf {
-		compatible = "renesas,mmngrbuf";
-	};
-
-	vspm_if {
-		compatible = "renesas,vspm_if";
-	};
-
-	reg_12p0v: regulator-12p0v {
-		compatible = "regulator-fixed";
-		regulator-name = "D12.0V";
-		regulator-min-microvolt = <12000000>;
-		regulator-max-microvolt = <12000000>;
-		regulator-boot-on;
-		regulator-always-on;
 	};
 
 	sound: sound {
@@ -153,62 +73,29 @@
 		compatible = "ti,pcm5102a";
 		#sound-dai-cells = <0>;
 	};
-
-	vcc_sdhi0: regulator-vcc-sdhi0 {
-		compatible = "regulator-fixed";
-
-		regulator-name = "SDHI0 Vcc";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-		regulator-boot-on;
-	};
-
-	vccq_sdhi0: regulator-vccq-sdhi0 {
-		compatible = "regulator-gpio";
-
-		regulator-name = "SDHI0 VccQ";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <3300000>;
-
-		gpios = <&gpio3 13 GPIO_ACTIVE_HIGH>;
-		gpios-states = <1>;
-		states = <3300000 1
-			  1800000 0>;
-	};
-
-	wlan_en_reg: fixedregulator {
-		compatible = "regulator-fixed";
-		regulator-name = "wlan-en-regulator";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		startup-delay-us = <70000>;
-
-		gpio = <&gpio2 25 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
-
-	x13_clk: x13 {
-		compatible = "fixed-clock";
-		#clock-cells = <0>;
-		clock-frequency = <74250000>;
-	};
 };
 
 &audio_clk_a {
 	clock-frequency = <22579200>;
 };
 
-&du {
-	pinctrl-0 = <&du_pins>;
-	pinctrl-names = "default";
-	status = "okay";
+&sound {
+	simple-audio-card,dai-link@1 {
+		reg = <1>;
+		format = "i2s";
+		cpu {
+			sound-dai = <&rcar_sound 1>;
+			bitclock-master;
+			frame-master;
+		};
 
-	clocks = <&cpg CPG_MOD 724>,
-		 <&cpg CPG_MOD 723>,
-		 <&x13_clk>;
-	clock-names = "du.0", "du.1", "dclkin.0";
+		codec {
+			sound-dai = <&tda19988>;
+		};
+	};
+};
 
+&du {
 	ports {
 		port@0 {
 			endpoint {
@@ -218,46 +105,6 @@
 	};
 };
 
-&ehci0 {
-	dr_mode = "host";
-	status = "okay";
-
-	memory-region = <&global_cma>;
-};
-
-&extal_clk {
-	clock-frequency = <48000000>;
-};
-
-&hscif2 {
-	pinctrl-0 = <&hscif2_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-
-	bluetooth {
-		compatible = "ti,wl1837-st";
-		enable-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
-	};
-};
-
-&hscif3 {
-	pinctrl-0 = <&hscif3_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-};
-
-&hscif4 {
-	pinctrl-0 = <&hscif4_pins>;
-	pinctrl-names = "default";
-
-	uart-has-rtscts;
-	status = "okay";
-};
-
 &i2c0 {
 	status = "okay";
 	clock-frequency = <100000>;
@@ -320,126 +167,6 @@
 	};
 };
 
-&i2c1 {
-	pinctrl-0 = <&i2c1_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-	clock-frequency = <400000>;
-
-	rtc@32 {
-		compatible = "epson,rx8571";
-		reg = <0x32>;
-	};
-};
-
-&lvds0 {
-	status = "okay";
-
-	clocks = <&cpg CPG_MOD 727>, <&x13_clk>, <&extal_clk>;
-	clock-names = "fck", "dclkin.0", "extal";
-};
-
-&ohci0 {
-	dr_mode = "host";
-	status = "okay";
-
-	memory-region = <&global_cma>;
-};
-
-&pcie_bus_clk {
-	clock-frequency = <100000000>;
-};
-
-&pciec0 {
-	/* Map all possible DDR as inbound ranges */
-	dma-ranges = <0x42000000 0 0x40000000 0 0x40000000 0 0x80000000>;
-};
-
-&pfc {
-	du_pins: du {
-		groups = "du_rgb888", "du_clk_out_0", "du_sync", "du_disp",
-			 "du_clk_in_0";
-		function = "du";
-	};
-
-	hscif2_pins: hscif2 {
-		groups = "hscif2_data_a", "hscif2_ctrl_a";
-		function = "hscif2";
-	};
-
-	hscif3_pins: hscif3 {
-		groups = "hscif3_data_c", "hscif3_ctrl_c";
-		function = "hscif3";
-	};
-
-	hscif4_pins: hscif4 {
-		groups = "hscif4_data_b";
-		function = "hscif4";
-	};
-
-	i2c1_pins: i2c1 {
-		groups = "i2c1_b";
-		function = "i2c1";
-	};
-
-	scif2_pins: scif2 {
-		groups = "scif2_data_a";
-		function = "scif2";
-	};
-
-	sdhi0_pins: sd0 {
-		groups = "sdhi0_data4", "sdhi0_ctrl";
-		function = "sdhi0";
-		power-source = <3300>;
-	};
-
-	sdhi0_pins_uhs: sd0_uhs {
-		groups = "sdhi0_data4", "sdhi0_ctrl";
-		function = "sdhi0";
-		power-source = <1800>;
-	};
-
-	sdhi3_pins: sd3 {
-		groups = "sdhi3_data4", "sdhi3_ctrl";
-		function = "sdhi3";
-		power-source = <1800>;
-	};
-
-	sound_clk_pins: sound_clk {
-		groups = "audio_clkout1_a",
-			 "audio_clk_a", "audio_clk_b_a", "audio_clk_c_a";
-		function = "audio_clk";
-	};
-
-	sound_pins: sound {
-		groups = "ssi01239_ctrl", "ssi0_data",
-			 "ssi349_ctrl", "ssi3_data", "ssi4_data";
-		function = "ssi";
-	};
-
-	usb30_pins: usb30 {
-		groups = "usb30", "usb30_id";
-		function = "usb30";
-	};
-
-	pwm6_pins: pwm6 {
-		groups = "pwm6_a";
-		function = "pwm6";
-	};
-
-	tpu0_pins: tpu {
-		groups = "tpu0_to0", "tpu0_to1";
-		function = "tpu0";
-	};
-
-	msiof0_pins: spi0 {
-		groups = "msiof0_clk", "msiof0_sync",
-			 "msiof0_rxd", "msiof0_txd";
-		function = "msiof0";
-	};
-};
-
 &rcar_sound {
 	pinctrl-0 = <&sound_pins &sound_clk_pins>;
 	pinctrl-names = "default";
@@ -462,67 +189,26 @@
 			playback = <&ssi4 &src4 &dvc1>;
 		};
 	};
+
 };
 
 &ssi4 {
 	shared-pin;
 };
 
-&rwdt {
-	timeout-sec = <60>;
-	status = "okay";
-};
-
-&scif2 {
-	pinctrl-0 = <&scif2_pins>;
-	pinctrl-names = "default";
-
-	status = "okay";
-};
-
-&sdhi0 {
-	pinctrl-0 = <&sdhi0_pins>;
-	pinctrl-1 = <&sdhi0_pins_uhs>;
-	pinctrl-names = "default", "state_uhs";
-
-	vmmc-supply = <&vcc_sdhi0>;
-	vqmmc-supply = <&vccq_sdhi0>;
-	cd-gpios = <&gpio3 12 GPIO_ACTIVE_LOW>;
-	bus-width = <4>;
-	sd-uhs-sdr50;
-	sd-uhs-sdr104;
-	status = "okay";
+&ohci0 {
+	dr_mode = "host";
 };
 
-&sdhi3 {
-	status = "okay";
-	pinctrl-0 = <&sdhi3_pins>;
-	pinctrl-names = "default";
-
-	vmmc-supply = <&wlan_en_reg>;
-	bus-width = <4>;
-	non-removable;
-	cap-power-off-card;
-	keep-power-in-suspend;
-
-	#address-cells = <1>;
-	#size-cells = <0>;
-	wlcore: wlcore@2 {
-		compatible = "ti,wl1837";
-		reg = <2>;
-		interrupt-parent = <&gpio1>;
-		interrupts = <0 IRQ_TYPE_LEVEL_HIGH>;
-	};
+&ehci0 {
+	dr_mode = "host";
 };
 
 &usb2_phy0 {
 	renesas,no-otg-pins;
-	status = "okay";
 };
 
 &usb3_peri0 {
-	companion = <&xhci0>;
-	status = "okay";
 	usb-role-switch;
 
 	port {
@@ -535,65 +221,40 @@
 &xhci0 {
 	pinctrl-0 = <&usb30_pins>;
 	pinctrl-names = "default";
-
-	status = "okay";
-
-	memory-region = <&global_cma>;
 };
 
-&cmt0 {
-	status = "okay";
-};
-
-&cmt1 {
-	status = "okay";
-};
-
-&tmu0 {
-	status = "okay";
-};
-
-&tmu1 {
-	status = "okay";
-};
-
-&tmu2 {
-	status = "okay";
-};
-
-&tmu3 {
-	status = "okay";
-};
-
-&tmu4 {
-	status = "okay";
-};
-
-&pwm6 {
-	pinctrl-0 = <&pwm6_pins>;
+&sdhi3 {
+	pinctrl-0 = <&sdhi3_pins>;
 	pinctrl-names = "default";
-
-	status = "okay";
 };
 
-&vspb {
-	status = "okay";
-};
-
-&vspi0 {
-	status = "okay";
+&hscif2 {
+	bluetooth {
+		enable-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
+	};
 };
 
-&tpu{
-	pinctrl-0 = <&tpu0_pins>;
-	pinctrl-names = "default";
+&pfc {
+	sdhi3_pins: sd3 {
+		groups = "sdhi3_data4", "sdhi3_ctrl";
+		function = "sdhi3";
+		power-source = <1800>;
+	};
 
-	status = "okay";
-};
+	usb30_pins: usb30 {
+		groups = "usb30", "usb30_id";
+		function = "usb30";
+	};
 
-&msiof0 {
-	pinctrl-0 = <&msiof0_pins>;
-	pinctrl-names = "default";
+	sound_clk_pins: sound_clk {
+		groups = "audio_clkout1_a",
+			 "audio_clk_a", "audio_clk_b_a", "audio_clk_c_a";
+		function = "audio_clk";
+	};
 
-	status = "okay";
+	sound_pins: sound {
+		groups = "ssi01239_ctrl", "ssi0_data",
+			 "ssi349_ctrl", "ssi3_data", "ssi4_data";
+		function = "ssi";
+	};
 };
-- 
2.7.4

